### DISCUSSION TITLE
[Question]: Locked while downloading?

### DISCUSSION THREAD

**Discussion by Woisek** [oobabooga/text-generation-webui#2747] [Q&A]
Simple question: Why is the whole interface locked, while downloading a model?
Why can't we just use it normally and chat while a model is downloaded? Makes no sense to my. Any logical explanation to that? 🤔

**Comment by BetaDoggo** [comment]
Gradio can't really handle multiple actions at once. Gradio won't do anything until the previous input has returned its output. Most other Gradio based interfaces have the same issue. Gradio has been extended far beyond its original purpose as a demo making tool.

**Reply by Woisek** [reply]
Oh man ... seems like Gradio is always in the way instead of helping us ... 🙄
And there is no way to handle the download "outside" of Gradio ... somehow? 

>Gradio won't do anything until the previous input has returned its output.

Maybe there is a way to trick Gradio for just this option? 🤔

Thanks for the response, btw ...

**Comment by viperwasp** [comment]
A lot of models can be downloaded without using the built in UI.

**Reply by Woisek** [reply]
Absolutely right, but it's tedious to click those many single files into a downloader. There could be one missed or so. The download in text-gen is so much convenient ... 🙂
That's also the reason why I asked here. Pure laziness ... 😋

**Comment by FartyPants** [comment]
I gutted the function and made a separate gradio that can be used in parallel
https://github.com/FartyPants/download_UI

**Reply by Woisek** [reply]
This is fantastic! Much appreciated, thanks! 👍🎉🎉
